codegen.erb

```diff
@@ -36,16 +36,16 @@ impl MethodFrame {
     }
 
     // We need this method, so we can match on class_id & method_id
-    pub fn decode(frame: Frame) -> MethodFrame {
+    pub fn decode(frame: Frame) -> AMQPResult<MethodFrame> {
         if frame.frame_type != FrameType::METHOD {
-            panic!("Not a method frame");
+            return Err(AMQPError::DecodeError("Not a method frame"))
         }
         let reader = &mut &frame.payload[..];
         let class_id = reader.read_u16::<BigEndian>().unwrap();
```

Comment:
Ouch, I didn't check for those.
It looks like the code is full of them (~290 from a quick grep). May I polish this PR first and attack them later?